crates/uv-distribution/src/metadata/requires_dist.rs

```diff
@@ -127,10 +127,23 @@ impl RequiresDist {
                 .flatten()
                 .collect::<BTreeMap<_, _>>();
 
+            // Get additional settings
+            let empty_settings = ToolUvDependencyGroups::default();
+            let group_settings = project_workspace
+                .current_project()
+                .pyproject_toml()
+                .tool
+                .as_ref()
+                .and_then(|tool| tool.uv.as_ref())
+                .and_then(|uv| uv.dependency_groups.as_ref())
+                .unwrap_or(&empty_settings);
+
             // Flatten the dependency groups.
-            let mut dependency_groups =
-                FlatDependencyGroups::from_dependency_groups(&dependency_groups)
-                    .map_err(|err| err.with_dev_dependencies(dev_dependencies))?;
+            let mut dependency_groups = FlatDependencyGroups::from_dependency_groups(
+                &dependency_groups,
+                group_settings.inner(),
+            )
```

Comment:
In earlier commits this is done in LowerRequirement::from_requirement as charlie suggested, but after implementing+testing I realized this is "too late" because of dependency-group `includes`, which gets erased during flattening, making us forget where requirements come from.